flexmeasures/cli/data_add.py

```diff
@@ -1117,6 +1125,293 @@ def add_schedule_for_storage(
+    reporter_config: TextIOBase,
+    start: Optional[datetime] = None,
+    end: Optional[datetime] = None,
+    start_offset: Optional[str] = None,
+    end_offset: Optional[str] = None,
+    resolution: Optional[timedelta] = None,
+    output_file: Optional[Path] = None,
+    dry_run: bool = False,
+    last_hour: bool = False,
+    last_day: bool = False,
+    last_7_days: bool = False,
+    last_month: bool = False,
+    last_year: bool = False,
+    timezone: str | pytz.BaseTzInfo = get_timezone(),
+):
+    """
+    Create a new report using the Reporter class and save the results
+    to the database or export them as csv or excel file.
+    """
+
+    # parse timezone into a BaseTzInfo object
+    if isinstance(timezone, str):
+        check_timezone(timezone)
+        timezone = pytz.timezone(zone=timezone)
+
+    # check that only 1 flag is provided
+    last_x_flags = [last_hour, last_day, last_7_days, last_month, last_year]
+    last_x_flag_given = last_x_flags.count(True) == 1
+
+    now = timezone.localize(datetime.now())
+
+    # if any of the last-X flag is provided, override start and end
+    if last_x_flag_given:
+        start, end = get_timerange_from_flag(
+            last_hour=last_hour,
+            last_day=last_day,
+            last_7_days=last_7_days,
+            last_month=last_month,
+            last_year=last_year,
+            timezone=timezone,
+        )
+
+    # apply offsets, if provided
+    if start_offset is not None:
+        if start is None:
+            start = now
+        start = apply_offset_chain(start, start_offset)
+
+    if end_offset is not None:
+        if end is None:
+            end = now
+        end = apply_offset_chain(now, end_offset)
+
+    # the case of not getting --start, --start-offset or any --last-X flag
+    if start is None:
+        click.secho(
+            "Either --start, --start-offset or any of the --last-X flags should be provided."
+            " Trying to use the latest datapoint of the report sensor as the start time and "
+            "the current time as the end...",
+            **MsgStyle.WARN,
+        )
+        last_value_datetime = (
+            db.session.query(func.max(TimedBelief.event_start))
+            .filter(TimedBelief.sensor_id == sensor.id)
+            .one_or_none()
+        )
+
+        # If there's data saved to the reporter sensors
+        if last_value_datetime[0] is not None:
+            start = last_value_datetime[0]
+        else:
+            click.secho(
+                f"Could not find any data for the report sensor {sensor}.",
+                **MsgStyle.ERROR,
+            )
+            raise click.Abort()
+
+    # the case of not getting --start, --start-offset or any --last-X flag
+    if end is None:
+        click.secho(
+            "Either --end, --end-offset or any of the --last-X flags should be provided."
+            " Trying to use the latest datapoint of the report sensor as the start time and "
```

Comment:
Inaccurate user info. If I only provide --start-offset, the latest datapoint of the report sensor is not used.